idaes/apps/grid_integration/multiperiod/tests/test_price_taker.py

```diff
@@ -0,0 +1,159 @@
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+
+from pathlib import Path
+import pytest
+import numpy as np
+import pandas as pd
+
+from pyomo.environ import (
+    Var,
+    Binary,
+    NonNegativeReals,
+    Constraint,
+    Expression,
+)
+
+from idaes.apps.grid_integration import DesignModel, OperationModel
+
+import idaes.logger as idaeslog
+
+from idaes.apps.grid_integration.multiperiod.price_taker_model import PriceTakerModel
+
+import matplotlib.pyplot as plt
+
+
+@pytest.fixture
+def excel_data():
+    DATA_DIR = Path(__file__).parent
+    file_path = DATA_DIR / "FLECCS.xlsx"
+    data = pd.read_excel(file_path, sheet_name=1)
+    return data
+
+
+@pytest.mark.unit
+def test_daily_data_size(excel_data):
+    m = PriceTakerModel()
+
+    # Generate price data for each hour of every day in the data
+    daily_data, scenarios = m.reconfigure_raw_data(excel_data)
+
+    # Check that there is a row for each horizon length in a representative day
+    assert len(daily_data) == m.horizon_length
+
+
+@pytest.mark.unit
+def test_determine_optimal_num_clusters(excel_data):
+    m = PriceTakerModel()
+
+    daily_data, scenarios = m.reconfigure_raw_data(excel_data)
+    n_clusters, inertia_values = m.get_optimal_n_clusters(daily_data)
+
+    assert n_clusters == 10
+
+
+@pytest.mark.unit
+def test_elbow_plot(excel_data):
+    m = PriceTakerModel()
+
+    daily_data, scenarios = m.reconfigure_raw_data(excel_data)
+    m.get_optimal_n_clusters(daily_data, plot=True)
+
+    assert plt.gcf() is not None
+
+
+@pytest.mark.unit
+def test_cluster_lmp_data(excel_data):
+    m = PriceTakerModel()
+
+    daily_data, scenarios = m.reconfigure_raw_data(excel_data)
+    n_clusters, inertia_values = m.get_optimal_n_clusters(daily_data)
+    lmp_data, weights = m.cluster_lmp_data(daily_data, n_clusters, scenarios)
+
+    sum_of_weights = 0
+    for i in range(0, n_clusters):
+        sum_of_weights = sum_of_weights + weights[0][i]
+    assert sum_of_weights == 365
+
+    assert len(lmp_data) == n_clusters
+
+
+@pytest.mark.unit
+def test_logger_messages(excel_data, caplog):
+    with caplog.at_level(idaeslog.WARNING):
+        m = PriceTakerModel()
+
+        daily_data, scenarios = m.reconfigure_raw_data(excel_data)
+        m.get_optimal_n_clusters(daily_data)
+
+        assert f"kmax was not set - using a default value of 30." in caplog.text
+
+    # TODO: The below test is not working because our data doesn't ever seem to arrive at an n_clusters close to kmax
```

Comment:
I couldn't get this test working for the reason stated in the TO-DO note. I've played around with it a bit and couldn't get it working. It _might_ be the case that we just need to generate a new dataset that is capable of arriving at an n_clusters close to kmax such that we can test this logger warning. Or maybe we should just leave this warning untested for now...